dsl/Minitest.cc

```diff
@@ -59,7 +59,7 @@ unique_ptr<ast::Expression> replaceDSLSingle(core::MutableContext ctx, ast::Send
         return nullptr;
     }
 
-    if (send->args.empty() && send->fun == core::Names::before()) {
+    if (send->args.empty() && (send->fun == core::Names::before() || send->fun == core::Names::after())) {
         return addSigVoid(ast::MK::Method0(send->loc, send->loc, core::Names::initialize(),
```

Comment:
(I don't entirely understand this – will this change mean it understands _two_ `initialize` methods? Is that OK?)

Reply:
Yeah we should not do this feature as implemented. It will mean sorbet will only check the contents of either the `before` or `after` block, whichever comes last.

Sorbet should probably treat an `after` block like it treats any other `it` block.

Reply:
Ack. It treats `it 'foo'` as `def test_foo`, which is how minitest actually handles it, but `after`s don't have a name, so it's a bit different. How would you recommend giving each `after` block (since there can be multiple) a unique name? (random string?)

Reply:
I don't know much C++, so this is spitballing, but what about something like, on L88,

```c++
    } else if (send->fun == core::Names::after()) {
        auto name = ctx.state.enterNameUTF8("<after_" + std::to_string(Random::uniformU4()) + ">");
        return addSigVoid(ast::MK::Method0(send->loc, send->loc, std::move(name),
                                           prepareBody(ctx, std::move(send->block->body)),
                                           ast::MethodDef::DSLSynthesized));
    }

```
❓

Reply:
(I realize that would be deterministic, so maybe that's bad. Maybe just increment a counter per file?)

Reply:
> It will mean sorbet will only check the contents of either the before or after block, whichever comes last.

(per slack convo) That means that currently, if there are multiple `before` blocks, it's only checking one of them? ... that's a bug.

Reply:
> Yeah we should not do this feature as implemented. It will mean sorbet will only check the contents of either the `before` or `after` block, whichever comes last.

Dmitry pointed out I was misremembering how infer decides which methods to typecheck (it iterates over MethodDef nodes, not over symbols in the symbol table) so this criticism isn't actually valid.